src/db.c

```diff
@@ -230,9 +232,9 @@ static void dbSetValue(redisDb *db, robj *key, robj *val, int overwrite) {
     serverAssertWithInfo(NULL,key,de != NULL);
     dictEntry auxentry = *de;
     robj *old = dictGetVal(de);
-    if (server.maxmemory_policy & MAXMEMORY_FLAG_LFU) {
```

Comment:
can you explain why this is redundant?
is it because it's a must in case of LFU, but on the other hand doesn't cause any harm in LRU?

I.e in LFU we wanna keep the existing value and increment it.
and in LRU we just refreshed one in lookupKey and a created another (fresh) one when creating the new string object we're replacing.
so with your change we'll keep the refreshed one from lookupKey.
is that right?
let's clearly describe it in the top comment.